include/envoy/stats/symbol_table.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "envoy/common/pure.h"
+
+namespace Envoy {
+namespace Stats {
+
+using Symbol = uint32_t;
+using SymbolVec = std::vector<Symbol>;
+
+/**
+ * Interface for storing a stat name, which is a wrapper around a SymbolVec.
+ */
+class StatName {
+public:
+  StatName() {}
+  virtual ~StatName(){};
+  virtual std::string toString() const PURE;
+  virtual SymbolVec toSymbols() const PURE;
+};
+
+using StatNamePtr = std::unique_ptr<StatName>;
```

Comment:
Should this be `SharedPtr` to support ref counted sharing?

Reply:
I don't think so -- StatNames should be totally unique, AFAIK.

Reply:
Yeah I had thought of this too and almost wrote the same comment. I don't think this needs to be shared, but it's possible, based on usage there might be emerge a need.